sklearn/utils/validation.py

```diff
@@ -1096,10 +1096,12 @@ def is_sparse(dtype):
                 "Convert your data to numeric values explicitly instead."
             )
         if not allow_nd and array.ndim >= 3:
-            raise ValueError(
-                "Found array with dim %d. %s expected <= 2."
-                % (array.ndim, estimator_name)
-            )
+            if estimator_name is not None:
+                raise ValueError(
+                    "Found array with dim %d. %s expected <= 2."
+                    % (array.ndim, estimator_name)
+                )
+            raise ValueError("Found array with dim %d. Expected <= 2." % (array.ndim))
```

Comment:
I think we already have something in place a few lines up where we define:
```
    estimator_name = _check_estimator_name(estimator)
    context = " by %s" % estimator_name if estimator is not None else ""
```

What about using the context variable in the error message? 
The advantage would be not to require an additional conditional check.

Apart from this (and disregarding the current way we do it), it would be nice to use f-strings for this, I think.